src/docx-etl.py

```diff
@@ -4,58 +4,94 @@
+sb_api_url = os.environ["SUPABASE_API_URL"]  # github actions secret management
+sb_api_key = os.environ["SUPABASE_API_KEY"]  # github actions secret management
 
-senate_rss_feed = "https://www.legis.state.pa.us/WU01/LI/RSS/SenateBills.xml"
-house_rss_feed = "https://www.legis.state.pa.us/WU01/LI/RSS/HouseBills.xml"
 
-sb_api_url = "https://vsumrxhpkzegrktbtcui.supabase.co"
-#sb_api_key = os.getenv("SUPABASE_API_KEY") #local secret management wtih .env file
-sb_api_key = os.environ["SUPABASE_API_KEY"] #github actions secret management 
-
-def dowload_bill_doc_file(bill_link):
-    bill_link = bill_link.replace("txtType=HTM", "txtType=DOC") #change link to doc download
+def download_bill_doc_file(bill_link: str) -> str:
+    """
+    Downloads a .doc file from a given link
+    :param bill_link: link to the .doc file
+    :return: path to the downloaded .doc file
+    """
+    bill_link = bill_link.replace("txtType=HTM", "txtType=DOC")  # change link to doc download
     response = requests.get(bill_link, stream=True)
     output_file_name = "bill.doc"
-    with open(output_file_name, "wb") as f: #add try catch here and return empty filname if it fails
+    with open(output_file_name, "wb") as f:  # add try catch here and return empty filename if it fails
         f.write(response.content)
     return output_file_name
 
-def convert_doc_to_docx(bill_path):
-    new_file_name = ""
+
+def convert_doc_to_docx(bill_path: str) -> str:
+    """
+    Converts a .doc file to a .docx file using LibreOffice
+    :param bill_path: path to the .doc file
+    :return: path to the .docx file
+    """
     try:
         subprocess.call(['libreoffice', '--headless', '--convert-to', 'docx', bill_path])
-        new_file_name = bill_path.replace(".doc", ".docx")
+        return bill_path.replace(".doc", ".docx")
     except FileNotFoundError:
         raise FileNotFoundError(f"The file could not be found.")
-    return new_file_name
 
-def extract_law_text_from_docx(bill_path):
-    output_text = ""
-    doc = Document(bill_path)
 
-    #update to only include docs that haven't been posted yet
-    # find a way to determine when the actual bill text starts
+def extract_law_text_from_docx(bill_path: str) -> str:
+    """
+    Extracts the law text from a .docx file
+    :param bill_path: path to the .docx file
+    :return: extracted law text
+    """
 
-    for para in doc.paragraphs:
-        for run in para.runs: #not sure what runs are, but these for loops seem to look through on a character by character basis (testing by strikethroughing a single character and this picks it up as an individual run)
-            if not run.font.strike: #only extract text that isn't strikethrough 
+    # TODO: update to only include docs that haven't been posted yet
+    # TODO: find a way to determine when the actual bill text starts
+    doc = Document(bill_path)
+    output_text = ""
+    for paragraph in doc.paragraphs:
```

Comment:
Please test to see if each paragraph ends with a newline (`\n` or `\r\n`), if not, we need to insert it after reach paragraph

Reply:
Good catch; added. This might not be terribly important since we'll be mostly using the raw text for analysis, but adding the line breaks doesn't hurt and can make it a bit more readable.